remme/clients/block_info.py

```diff
@@ -23,29 +23,27 @@ class BlockInfoClient(BasicClient):
         return bi
 
     async def get_blocks_info(self, start, limit):
-        if not (start and limit):
-            block_config = None
-            try:
-                block_config = await self.get_block_info_config()
-            except Exception:
-                return []
+        init_start = start or None
+        init_limit = limit or None
 
-            if not start:
-                start = block_config.latest_block + 1
+        try:
+            block_config = await self.get_block_info_config()
+        except Exception:
+            return []
 
-            if not limit:
-                limit = start - block_config.oldest_block + 1
+        start = block_config.oldest_block if not init_start else start - 1
 
-        if limit - start > 0:
-            limit = start
+        limit = block_config.latest_block if not init_limit else limit - 1
+
+        end = min(start + limit, block_config.latest_block, 100) + 1
```

Comment:
Explain this line, please.

For example:

* We want to get blocks from 1000 to 1200 while we have 2300 blocks in total;
* We are limited by 100 blocks, so we must receive blocks from 1000 to 1100;
* However, this line will give us: `min(1000 + 200, 2300, 100) = 100`.

Isn't that wrong?